client/components/student/student.service.js

```diff
@@ -0,0 +1,14 @@
+'use strict';
+
+var app = angular.module('app');
+
+app.factory('Student', function($resource) {
+  return $resource('/api/students/:id', {
+    id: '@_id'
+  }, {
+    list: {
+      method: 'GET',
+      isArray: true
+    }
+  });
```

Comment:
https://docs.angularjs.org/api/ngResource/service/$resource

`$resource` is a convenience wrapper for common rest request types. The second object contains custom calls.

To use the `list` function of the `$resource` we would just call `Student.list()`. Since we don't pass in an `id` the call would be made to `/api/students/` which is the path to return all students.